bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/util/DynamicContextBuilder.java

```diff
@@ -144,9 +146,22 @@ public URI resolveUri(String uri) {
 			if (realURI.isAbsolute()) {
 				return realURI;
 			}
+
+			if (_staticContext.getBaseUri() == null) {
+				throw DynamicError.noBaseURI();
```

Comment:
:warning: This is assuming we want FONS0005 returned. The documentation for `fn:doc` mentions the use of `fn:resolve-uri` which does produce this error, but the documentation of `fn:doc` does not directly mention FONS0005 as a possibility. The test suite does not appear to cover this case specifically, and it would only be expected to arise when the user fails to set the base URI for the static context.